internal/controller/build/controller.go

```diff
@@ -452,7 +452,11 @@ func (r *Reconciler) fetchComponentConfigs(ctx context.Context, buildCtx *integr
 		logger.Error(err, "Failed to fetch component descriptor")
 		return nil, fmt.Errorf("failed to get component.yaml from the repository buildName:%s;%w", buildCtx.Build.Name, err)
 	}
-	return config.(*source.Config), nil
+	sourceConfig, ok := config.(source.Config)
```

Comment:
```suggestion
	sourceConfig, ok := config.(*source.Config)
```

wouldn't this fail if the config is actually *source.Config

Reply:
config returned from the function call is not a pointer. So this doesn't allow to convert to type "config.(*source.Config)". Instead, we have to convert it to "config.(source.Config)".

Reply:
@Mirage20 What do you think?

Reply:
@VajiraPrabuddhaka changed the interface